core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageSerializationConfigStep.java

```diff
@@ -0,0 +1,59 @@
+package io.quarkus.deployment.steps;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import io.quarkus.builder.Json;
+import io.quarkus.builder.Json.JsonArrayBuilder;
+import io.quarkus.builder.Json.JsonObjectBuilder;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.GeneratedResourceBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.LambdaCapturingTypeBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+
+public class NativeImageSerializationConfigStep {
+
+    @BuildStep
```

Comment:
Shouldn't we use `@BuildStep(onlyIf = NativeOrNativeSourcesBuild.class)` instead?

Reply:
It makes sense, but I am not sure. The `generateFeature` (which generates the native image feature previously used for the same reason) in `NativeImageFeatureStep` doesn't use this `onlyIf`.

Reply:
Actually non of the build steps named `io.quarkus.deployment.steps.NativeImage*` seems to come with this condition.

Reply:
> It makes sense, but I am not sure. The `generateFeature` (which generates the native image feature previously used for the same reason) in `NativeImageFeatureStep` doesn't use this `onlyIf`.

Yes, because it produces `GeneratedNativeImageClassBuildItem` and `JPMSExportBuildItem` which are only consumed when a native image is built. 

Here you produce the `GeneratedResourceBuildItem` which is also consumed when the fast-jar is generated... You can try to run some app in the dev mode and check the [build steps report](https://github.com/quarkusio/quarkus/blob/main/TROUBLESHOOTING.md#analysing-build-steps-execution-time) if the build step is executed.

Reply:
Thanks for the clarifications! This is now fixed.